lib/pundit.rb

```diff
@@ -36,7 +36,9 @@ def initialize(options = {})
         @record = options[:record]
         @policy = options[:policy]
 
-        message = options.fetch(:message) { "not allowed to #{query} this #{record.class}" }
+        record_name = record.is_a?(Class) ? record.name : "this #{record.class}"
+
+        message = options.fetch(:message) { "not allowed to #{policy.class}##{query} #{record_name}" }
```

Comment:
Hi! I've thought about this some more and I've got a change that's worth making.

The reasons for the change are:
1. `record.to_s` works better for anonymous classes, and it's what we use in PolicyFinder so it's somewhat battle-tested.
2. If there's an explicit message we should not calculate a record name.

```suggestion
        message = options.fetch(:message) do
          record_name = record.is_a?(Class) ? record.to_s : "this #{record.class}"
          "not allowed to #{policy.class}##{query} #{record_name}"
        end
```